File: AtlasWeb/Core/Modules/Tibbi_Surec/Sosyal_Hizmetler_Modulu/SocialServicesPatientInterviewFormViewModel.cs
//$DC37CF3E
using System;
using System.Linq;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class PatientInterviewFormServiceController
    {
        partial void PreScript_SocialServicesPatientInterviewForm(SocialServicesPatientInterviewFormViewModel viewModel, PatientInterviewForm patientInterviewForm, TTObjectContext objectContext)
        {
            /*Guid? selectedEpisodeActionObjectID = viewModel.GetSelectedEpisodeActionID();
            if (viewModel._PatientInterviewForm.Episode == null)
            {
                if (selectedEpisodeActionObjectID.HasValue)
                {
                    EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionObjectID.Value);
                    viewModel._PatientInterviewForm.Episode = episodeAction.Episode;
                    viewModel._PatientInterviewForm.MasterAction = episodeAction;
                    viewModel.ResUsers = patientInterviewForm.ObjectContext.LocalQuery<ResUser>().ToArray();
                }
            }*/
            if(patientInterviewForm.Episode.Patient.SKRSMaritalStatus != null && viewModel._PatientInterviewForm.MaritalStatus == null)
            {
                //todo my 
                //viewModel._PatientInterviewForm.MaritalStatus = patientInterviewForm.Episode.Patient.SKRSMaritalStatus.ADI;
            }
            if(patientInterviewForm.SubEpisode.PatientAdmission.PA_Address.MobilePhone != null && viewModel._PatientInterviewForm.PatientPhoneNum == null)
            {
                viewModel._PatientInterviewForm.PatientPhoneNum = patientInterviewForm.SubEpisode.PatientAdmission.PA_Address.MobilePhone;
            }
            if (patientInterviewForm.SubEpisode.PatientAdmission.PA_Address.HomeAddress != null && viewModel._PatientInterviewForm.PatientAddress == null)
            {
                viewModel._PatientInterviewForm.PatientAddress = patientInterviewForm.SubEpisode.PatientAdmission.PA_Address.HomeAddress;
            }

            DiagnosisGrid[] episodeDiagnosis = patientInterviewForm.Episode.Diagnosis.ToArray();
            if (episodeDiagnosis.Length > 0)
            {
                foreach (DiagnosisGrid dG in episodeDiagnosis)
                {
                    viewModel.Diagnosis +=  dG.Diagnose.Name + "-- ";
                }
                viewModel.Diagnosis = viewModel.Diagnosis.Remove(viewModel.Diagnosis.Length - 3);
                if(patientInterviewForm.SocServPatientFamilyInfo != null)
                    patientInterviewForm.SocServPatientFamilyInfo.PatientDiagnosis = viewModel.Diagnosis;

            }
            if(patientInterviewForm.Episode.Payer != null)
            {
                viewModel.PayerName = patientInterviewForm.Episode.Payer.Name.ToString();
            }
            if (patientInterviewForm.SubEpisode.InpatientAdmission!= null && patientInterviewForm.SubEpisode.InpatientAdmission.Bed != null)
            {
                viewModel.RoomName = patientInterviewForm.SubEpisode.InpatientAdmission.Bed.Room.Name;
            }
            else
            {
                viewModel.RoomName = "";
            }
            CompanionApplication companionApplication = patientInterviewForm.Episode.GetActiveCompanionApplication();
            
            if (companionApplication != null)
            {
                viewModel.CompanionName = companionApplication.CompanionName;
                //viewModel.CompanionPhoneNumber = companionApplication.CompanionPhoneNum;

                if(viewModel._PatientInterviewForm.SocServPatientFamilyInfo != null)
                { 
                    viewModel._PatientInterviewForm.SocServPatientFamilyInfo.Companion = companionApplication.CompanionName;
                    //viewModel._PatientInterviewForm.SocServPatientFamilyInfo.CompanionPhoneNumber = companionApplication.CompanionPhoneNum;
                }
            }


            ContextToViewModel(viewModel, objectContext);
            if (viewModel._PatientInterviewForm.ExaminationDate == null)
                viewModel._PatientInterviewForm.ExaminationDate = Common.RecTime();
        }

        partial void PostScript_SocialServicesPatientInterviewForm(SocialServicesPatientInterviewFormViewModel viewModel, PatientInterviewForm patientInterviewForm, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            patientInterviewForm.WorkListDate = Common.RecTime();
        }
    }
}

namespace Core.Models
{
    public partial class SocialServicesPatientInterviewFormViewModel
    {
        public string CompanionName;
        public string CompanionPhoneNumber;
        public string RoomName;
        public string Diagnosis;
        public string PayerName;
        public ResUser ReportSelectedUser { get; set; }

    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Sosyal_Hizmetler_Modulu/SocialServicesPatientInterviewFormViewModel.ts
//$0D942CE7
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { PatientInterviewForm } from 'NebulaClient/Model/AtlasClientModel';
import { ResSection } from 'NebulaClient/Model/AtlasClientModel';
import { ResUser } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSMeslekler } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSOgrenimDurumu } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSMedeniHali } from 'NebulaClient/Model/AtlasClientModel';
import { SocServAppliedProcedures } from 'NebulaClient/Model/AtlasClientModel';
import { SocServPatientFamilyInfo } from 'NebulaClient/Model/AtlasClientModel';
import { Episode } from 'NebulaClient/Model/AtlasClientModel';
import { ResRoom } from 'NebulaClient/Model/AtlasClientModel';
import { Patient } from 'NebulaClient/Model/AtlasClientModel';
import { PatientIdentityAndAddress } from 'NebulaClient/Model/AtlasClientModel';
import { PayerDefinition } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';

export class SocialServicesPatientInterviewFormViewModel extends BaseViewModel {
    @Type(() => PatientInterviewForm)
    public _PatientInterviewForm: PatientInterviewForm = new PatientInterviewForm();
    @Type(() => ResSection)
    public ResSections: Array<ResSection> = new Array<ResSection>();
    @Type(() => ResUser)
    public ResUsers: Array<ResUser> = new Array<ResUser>();
    @Type(() => SKRSMeslekler)
    public SKRSMesleklers: Array<SKRSMeslekler> = new Array<SKRSMeslekler>();
    @Type(() => SKRSOgrenimDurumu)
    public SKRSOgrenimDurumus: Array<SKRSOgrenimDurumu> = new Array<SKRSOgrenimDurumu>();
    @Type(() => SKRSMedeniHali)
    public SKRSMaritalStatuss: Array<SKRSMedeniHali> = new Array<SKRSMedeniHali>();
    @Type(() => SocServAppliedProcedures)
    public SocServAppliedProceduress: Array<SocServAppliedProcedures> = new Array<SocServAppliedProcedures>();
    @Type(() => SocServPatientFamilyInfo)
    public SocServPatientFamilyInfos: Array<SocServPatientFamilyInfo> = new Array<SocServPatientFamilyInfo>();
    @Type(() => Episode)
    public Episodes: Array<Episode> = new Array<Episode>();
    @Type(() => ResRoom)
    public ResRooms: Array<ResRoom> = new Array<ResRoom>();
    @Type(() => Patient)
    public Patients: Array<Patient> = new Array<Patient>();
    @Type(() => PatientIdentityAndAddress)
    public PatientIdentityAndAddresss: Array<PatientIdentityAndAddress> = new Array<PatientIdentityAndAddress>();
    @Type(() => PayerDefinition)
    public PayerDefinitions: Array<PayerDefinition> = new Array<PayerDefinition>();
    @Type(() => ResUser)
    ReportSelectedUser: ResUser = new ResUser();

    public CompanionName: string;
    public CompanionPhoneNumber: string;
    public RoomName: string;
    public PayerName: string;
    public Diagnosis: string;
}
